### ISSUE TITLE
Transect setup - Clicking Finish does not warn that your transect alterations are not saved (also consider workflow)

### ISSUE THREAD

**Issue by andrewvanbreda** [BiologicalRecordsCentre/SPRING#53]
Hi @DavidRoy 
When editing a transect, the system doesn't save (or warn you) that transect alterations won't be saved if you have forgotten to click Save Transect.

There are a few things about this that makes the situation a bit unclear.

I have tried to do a comparison with the behaviour with EBMS.
However I find EBMS a bit difficult to check because I don't have experience with it, and its behaviours are a bit on the non-standard side.
For instance, if I create a new transect, it doesn't even appear in my sites grid at all. However it is available to create a walk with.

I notice if I masquerade as you, you do have a test transect, but one that seems to be in a read-only mode.

So I tried editing my one by going directly to the URL (an action obviously not to be done normally, but I needed to see the confirmation message behaviour).
I noticed that EBMS does the same thing, if you click Finish there is no message to confirm you haven't saved, and no save occurs either, alterations are lost.
However I am not sure if you are supposed to be able to edit existing ones on EBMS, so that might be an unrealistic situation.

So my questions are
1. At what point in EBMS/SPRING is the transect supposed to be read only? I thought it might be after a walk is done, however if I go to the URL for the transect edit page after a walk is done, it is still editable.
I can try to work this out in the code, but always a bit difficult to see exactly with lots of rules.
2. Why can't I see it on my grid?
3. If it is supposed to allow editing before walks are done? if so, is that message problem I mentioned acceptable?

Sorry for the confusing message, but that is mainly because I am confused.

If you don't know the answers, I could try and phone Gary later. I don't appear to be able to tag him or assign him to issues for SPRING on Github.

**Comment by andrewvanbreda** [comment]
Note to myself from David about permissions from https://github.com/BiologicalRecordsCentre/SPRING/issues/35

"the logic is that the transect can only be edited by the creator. I also, think it get locked once setup in eBMS"

**Comment by DavidRoy** [comment]
@andrewvanbreda best to ask Gary I think.  In answer to your questions
1. Read-only once 'Finish' is selected
2. Which 'grid'?  The 'my-sites' page?
3. It's not so crucial to make 'read-only' for SPRING at this stage.  The eBMS is slightly different in that we are trying to avoid multiple transects being setup by recorders who are unaware that a site is already covered.  SPRING should be more clear-cut as it has the hierarchy of: 1km site - transect and the expectation of a single transect per 1km

**Comment by andrewvanbreda** [comment]
@DavidRoy Yes it was the my-sites page. I will ask Gary why EBMS does that. I might be able to my read-only without his assistance, but good to know not live critical

**Comment by andrewvanbreda** [comment]
@DavidRoy 
I spoke to Gary this evening.

Firstly, as the forms are evolving, and I think new ones are being developed for EBMS, he isn't surprised by the bug I noticed where clicking Finish without click Save Route can lose a change.
I am not quite sure how best to fix tht yet, but if EBMS is live with the same bug I assume it doesn't need a fix for SPRING go live (although obviously will try to fix as soon as I can)

The other thing we discussed was about the nature of data locking on EBMS and how this can be translated by SPRING.
The first thing to note is EBMS doesn't actually do anything special when Finish is clicked.
I think having spoke to Gary about this, the behaviour we want might simple be that you can edit your transect site until a walk is done, then it becomes read only (and an administrator can always edit if needed).
Again not needed for go live as previously mentioned.
What do you think?

**Comment by andrewvanbreda** [comment]
@DavidRoy 
Confirmation added to finish button
"Are you sure you wish to leave this page? (please make sure you save your route changes first if you have not done so already)"

**Comment by DavidRoy** [comment]
This issue can be closed and we can revisit the permission/locking logic if further issues arrise
